lib/net/ssh/authentication/certificate.rb

```diff
@@ -34,7 +34,12 @@ def self.read_certblob(buffer, type)
           # 0x20c49ba5e353f7 = 0x7fffffffffffffff/1000, the largest value possible for JRuby
           # JRuby Time.at multiplies the arg by 1000, and then stores it in a signed long.
           # 0x20c49ba5e353f7 = 292278994-08-17 01:12:55 -0600
-          cert.valid_before = Time.at([0x20c49ba5e353f7, buffer.read_int64].min)
+          if RUBY_PLATFORM == "java"
```

Comment:
sorry to be picky, but can you please put comment also inside if. 
I generally don't like comments as they often live a separate life from the code, keeping them closer together might help.

https://thecodinglove.com/code-comments

Reply:
Great idea. Fixed. I also don't like comments like this, but decided that it was needed to explain the magic number of `0x20c49ba5e353f7`.